Sources/Commands/SwiftPackageTool.swift

```diff
@@ -882,39 +882,48 @@ extension SwiftPackageTool {
                 help: "Target(s) to which the plugin command should be applied")
         var targetNames: [String] = []
 
-        @Argument(help: "Name of the command plugin to invoke")
-        var command: String
+        @Flag(name: .customLong("list"),
+              help: "List the available plugin commands")
+        var listCommands: Bool = false
 
-        @Argument(help: "Any arguments to pass to the plugin")
-        var arguments: [String] = []
-        
-        func findPlugins(matching command: String, in graph: PackageGraph) -> [PluginTarget] {
-            // Find all the available plugin targets.
-            let availablePlugins = graph.allTargets.compactMap{ $0.underlyingTarget as? PluginTarget }
-            
-            // Find and return the plugins that match the command.
-            return availablePlugins.filter {
-                // Filter out any non-command plugins and any whose verb is different.
-                guard case .command(let intent, _) = $0.capability else { return false }
-                return command == intent.invocationVerb
-            }
-        }
+        @Argument(parsing: .unconditionalRemaining,
+                  help: "Name and arguments of the command plugin to invoke")
+        var pluginCommand: [String] = []
 
         func run(_ swiftTool: SwiftTool) throws {
             // Load the workspace and resolve the package graph.
             let packageGraph = try swiftTool.loadPackageGraph()
-            
-            // Find the plugins that match the command.
-            swiftTool.observabilityScope.emit(info: "Finding plugin for command '\(command) '")
-            let matchingPlugins = findPlugins(matching: command, in: packageGraph)
+
+            // List the available plugins, if asked to.
+            if listCommands {
+                let allPlugins = availableCommandPlugins(in: packageGraph)
+                for plugin in allPlugins.sorted(by: { $0.name < $1.name }) {
+                    guard case .command(let intent, _) = plugin.capability else { return }
```

Comment:
nit: use filter?

Reply:
I wasn't sure how to do that while also unpacking the intent and permissions (which aren't yet used in this PR but should be).  How would that look?  I'd like to switch to filter if I can.